### ISSUE TITLE
Set up for extenders

### ISSUE THREAD

**Issue by dschaefer** [eclipse-cdt/cdt-gdb-adapter#50]
At QNX, we're extending the adapter to add a couple of quirks we have with our version of gdb. To do that we extend the backend and the debug session classes.

I'm just wondering how to set up the npm package so we can import those classes while maintaining the webpack bundled build output.

Which opens up my first question. What is the index.js used for as we have it now. It returns the path to the debugAdapter.js file. What is that used for? And why can't those things do the same as we do in vscode, the path is well known inside the package.

I would then use the index.js file to export the API to the debug adapter so we can simply import the classes from the module.

Thoughts?

**Comment by dschaefer** [comment]
BTW, I'd also like to change 'out' to 'dist' to be more "standards" compliant in the npm world. 'out' is more a vscode "standard".

**Comment by dschaefer** [comment]
I'll probably need to create an api.ts or something like that for the API so that the session doesn't start up.

**Comment by dschaefer** [comment]
Cool I'll use the "main" for the API then.

We'll have a .npmignore file that will determine what goes in the package. So what's in the dist directory could still be filtered out. I'd like to filter out the src directory for sure since that shouldn't be necessary to build against the adapter or run it in a client.

That's the beauty of yarn link so I can hook in a development version of the package while I'm working with it, and use the production version when I'm working on other things or doing our own production builds. But I'll have more firm opinions once I spend more quality time with it :).

**Comment by simark** [comment]
> We'll have a .npmignore file that will determine what goes in the package. So what's in the dist directory could still be filtered out. I'd like to filter out the src directory for sure since that shouldn't be necessary to build against the adapter or run it in a client.

Just my 2 cents: I find it useful when packages include their map and source (ts) files, so we can debug it easily if needed (without having to build it ourselves).

**Comment by dschaefer** [comment]
> > We'll have a .npmignore file that will determine what goes in the package. So what's in the dist directory could still be filtered out. I'd like to filter out the src directory for sure since that shouldn't be necessary to build against the adapter or run it in a client.
> 
> Just my 2 cents: I find it useful when packages include their map and source (ts) files, so we can debug it easily if needed (without having to build it ourselves).

Yeah, not sure I want that hopefully rare case to drive our production builds. Especially when this thing takes about 10 seconds to build. The hope is that we'll have a lot of clients (had an interesting inquiry on that topic today) and the size of the package on npm is going to matter.

**Comment by dschaefer** [comment]
So at the end of the day, I'm abandoning webpack for the adapter. Too hard to make it do what I want and tired of the fight :).
